mlir/tools/mlir-tblgen/AttrOrTypeDefGen.cpp

```diff
@@ -1023,6 +1023,55 @@ bool DefGenerator::emitDefs(StringRef selectedDialect) {
   return false;
 }
 
+//===----------------------------------------------------------------------===//
+// Type Constraints
+//===----------------------------------------------------------------------===//
+
+static const char *const typeConstraintDecl = R"(
+::llvm::LogicalResult {0}(::mlir::Type type);
+)";
+
+static const char *const typeConstraintDef = R"(
+::llvm::LogicalResult {0}(::mlir::Type type) {
```

Comment:
Why return LogicalResult? Seems like all of the current examples are written as if I'd expect a bool: "isValid<...>". Should we just return bool for these?